pkg/virt-handler/vm.go

```diff
@@ -2691,9 +2691,18 @@ func (d *VirtualMachineController) finalizeMigration(vmi *v1.VirtualMachineInsta
 	return nil
 }
 
+func vmiHasTerminationGracePeriod(vmi *v1.VirtualMachineInstance) bool {
+	return vmi.Spec.TerminationGracePeriodSeconds != nil && *vmi.Spec.TerminationGracePeriodSeconds != 0
+}
+
+func domainHasGracePeriod(domain *api.Domain) bool {
+	return domain != nil &&
+		domain.Spec.Metadata.KubeVirt.GracePeriod != nil &&
+		domain.Spec.Metadata.KubeVirt.GracePeriod.DeletionGracePeriodSeconds != 0
+}
+
```

Comment:
due to default grace period of 30 seconds, we should consider the grace period is set when it's  `nil` or `!= 0`